lib/resolve-config.js

```diff
@@ -40,5 +43,41 @@ module.exports = (
         ? CI_API_V4_URL
         : urlJoin(defaultedGitlabUrl, isNil(userGitlabApiPathPrefix) ? '/api/v4' : userGitlabApiPathPrefix),
     assets: assets ? castArray(assets) : assets,
+    proxy: getProxyConfiguration(userGitlabUrl, HTTP_PROXY, HTTPS_PROXY),
   };
 };
+
+function getProxyConfiguration(gitlabUrl, HTTP_PROXY, HTTPS_PROXY) {
+  const proxy = undefined;
+  if (HTTP_PROXY && gitlabUrl.startsWith('http://')) {
+    return {
+      agent: {
+        http: new HttpProxyAgent({
+          keepAlive: true,
+          keepAliveMsecs: 1000,
+          maxSockets: 256,
+          maxFreeSockets: 256,
+          scheduling: 'lifo',
+          proxy: HTTP_PROXY,
+        }),
+      },
+    };
+  }
+
+  if (HTTPS_PROXY && gitlabUrl.startsWith('https://')) {
+    return {
+      agent: {
+        https: new HttpsProxyAgent({
+          keepAlive: true,
+          keepAliveMsecs: 1000,
+          maxSockets: 256,
+          maxFreeSockets: 256,
+          scheduling: 'lifo',
+          proxy: HTTPS_PROXY,
+        }),
+      },
+    };
+  }
+
+  return proxy;
```

Comment:
If the user _doesn't_ define an `HTTP_PROXY` or `HTTPS_PROXY` variable, won't returning `undefined` here cause issues when we attempt to spread the `proxy` variable on line 73 of `proxy.js`?  I haven't tested this, but I assume this would cause something like this:

```
> ...undefined
> Uncaught SyntaxError: expected expression, got '...'
```

Reply:
If so, I think the easy fix is to just return an empty object here:

```suggestion
  return {};
```

Reply:
I made tests with `undefined` and `{}` values, and the both works. I agree this is better to return {} so I made the update.